### PR TITLE
devmodel file support

### PR DESCRIPTION
**Pull request by giggsoff**
The ability to provide file with devmodel overwrites.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Comment by rvs** [comment]
This looks reasonable, but I wondering is this ztype -> ptype change is going to trip zedcloud controller. @eriknordmark WDYT?

**Review by rvs — APPROVED** [review]
LGTM

File: cmd/edenConfig.go [hunk 1]
```diff
@@ -55,6 +56,19 @@ func reloadConfigDetails() {
...
 		devModel = viper.GetString("eve.devmodel")
 		eveRemote = viper.GetBool("eve.remote")
+		modelFile = viper.GetString("eve.devmodelfile")
+		if modelFile != "" {
+			filePath, err := filepath.Abs(modelFile)
+			if err != nil {
+				log.Errorf("cannot get absolute path for devmodelfile (%s): %v", modelFile, err)
+				filePath = ""
+			} else
+			if _, err := os.Stat(filePath); err != nil {
```
Comment (eriknordmark) [inline cmd/edenConfig.go:66]: Doesn't yetus complain about this style? "else if" on a single line is what I'd expect.

File: cmd/edenConfig.go [hunk 2]
```diff
@@ -55,6 +56,19 @@ func reloadConfigDetails() {
 		qemuFileToSave = utils.ResolveAbsPath(viper.GetString("eve.qemu-config"))
 		devModel = viper.GetString("eve.devmodel")
 		eveRemote = viper.GetBool("eve.remote")
+		modelFile = viper.GetString("eve.devmodelfile")
+		if modelFile != "" {
+			filePath, err := filepath.Abs(modelFile)
+			if err != nil {
+				log.Errorf("cannot get absolute path for devmodelfile (%s): %v", modelFile, err)
```
Comment (eriknordmark) [inline cmd/edenConfig.go:63]: Shouldn't this error be more than a log? log.Fatalf like the other issues with arguments?

File: cmd/edenConfig.go [hunk 3]
```diff
@@ -55,6 +56,19 @@ func reloadConfigDetails() {
...
+				filePath = ""
+			} else
+			if _, err := os.Stat(filePath); err != nil {
+				log.Errorf("cannot parse devmodelfile (%s): %v", modelFile, err)
```
Comment (eriknordmark) [inline cmd/edenConfig.go:67]: Ditto

**Pull request merged**